sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -491,13 +491,10 @@ case class RenameTable(
  */
 case class ShowTables(
     namespace: LogicalPlan,
-    pattern: Option[String],
-    override val output: Seq[Attribute] = ShowTables.OUTPUT) extends Command {
+    pattern: Option[String]) extends Command {
   override def children: Seq[LogicalPlan] = Seq(namespace)
-}
 
-object ShowTables {
-  val OUTPUT = Seq(
```

Comment:
Can we simply use `def` here? Then every `ShowTables` node will get a fresh output, while the output is super stable within the corresponding `ShowTables` node.

Reply:
This just introduces unnecessary boilerplate code. I don't see any benefits of that. Could you explain the motivation of having the `ShowTables` object with a method?

Reply:
All the leaf nodes that produce output put `output` in the constructor, so that the node won't change its output attr ids during plan transformation/copy. What's wrong with the existing approach (like ds v1/v2 scan nodes, Range, etc.) and why not follow it here?

Reply:
> so that the node won't change its output attr ids during plan transformation/copy.

Now, it `output` won't change too.

> What's wrong with the existing approach ...

I have the same question. What's wrong with existing approach before the PR https://github.com/apache/spark/pull/31245/, and why you had to introduce companion objects like `object ShowTables`?

Reply:
Since `copy()` invokes the default constructor and re-creates vals, I reverted the companion object and renamed `OUTPUT` to `getOutputAttrs` (because it is a method now, and it should conform to naming conventions).